### PR TITLE
Minor style and typo fixes

### PR DESCRIPTION
**Pull request by apjanke**
This PR contains various minor style and typo fixes. These are all trivial, IMHO, so they're rolled into a single commit.

Reflect Jira changing color of "Create" button.
Consistently style "GitHub" with internal caps.
Fix various typos.
Add link to "user mailing list" in Introduction.
Style "fault-tolerant" consistently, with hyphen.
Fix missing "of" in "Powered By".
Use curly quotes in "Powered By".
Style "e-commerce" consistently, with hyphen.
Ecosystem: use a comma after "e.g." for consistency. Fix some typos. Fix link to FastR (was pointing at the fastr-flink repo).
Use Unicode ellipsis character ("…") instead of three dots ("...").
Use Oxford commas consistently.

### CODE REVIEW

File: community.md [hunk 1]
```diff
@@ -3,7 +3,7 @@ title: "Community & Project Info"
...
+There are many ways to get help from the Apache Flink community. The [mailing lists](#mailing-lists) are the primary place where all Flink committers are present. If you want to talk with the Flink committers and users in a chat, there is an [IRC channel](#irc). Some committers are also monitoring [Stack Overflow](http://stackoverflow.com/questions/tagged/flink). Please remember to tag your questions with the *[flink](http://stackoverflow.com/questions/tagged/flink)* tag. Bugs and feature requests can either be discussed on the *dev mailing list* or on [JIRA]({{ site.jira }}). Those interested in contributing to Flink should check out the [contribution guide](how-to-contribute.html).
```
Comment (fhueske) [inline community.md]: Can you update the Stack Overflow tag to `apache-flink`?

File: contribute-code.md [hunk 2]
```diff
@@ -14,25 +14,25 @@ This document contains everything you need to know about contributing code to Ap
...
+If the description of a JIRA issue indicates that its resolution will touch sensible parts of the code base, be sufficiently complex, or add significant amounts of new code, the Flink community might request a design document. (Most contributions should not require a design document.) The purpose of this document is to ensure that the overall approach to address the issue is sensible and agreed upon by the community. JIRA issues that require a design document are tagged with the **`requires-design-doc`** label. The label can be attached by any community member who feels that a design document is necessary. A good description helps to decide whether a JIRA issue requires a design document or not. The design document must be added or attached to or link from the JIRA issue and cover the following aspects:
```
Comment (fhueske) [inline contribute-code.md]: `link` -> `linked` (last sentence)?

File: community.md (hunk 1 above)
Comment (apjanke) [reply community.md]: Updated.

File: contribute-code.md (hunk 2 above)
Comment (apjanke) [reply contribute-code.md]: Updated.

**Comment by fhueske** [comment]
Thanks for the update!
I'll merge this PR.

**Comment by apjanke** [comment]
:+1:

**Pull request closed without merging**